File: data structures and algorithms/cpp/08_036.cpp
#include <unordered_map>
#include <vector>

bool hasFrequencyThreshold(std::vector<int>& nums, int k) {
    std::unordered_map<int, int> counter;
    
    for (int num : nums) {
        counter[num]++;
        if (counter[num] >= k) {
            return true;
        }
    }
    
    return false;
}

File: data structures and algorithms/cs/08_036.cs
using System.Collections.Generic;

public class FrequencyThresholdChecker
{
   public static bool HasFrequencyThreshold(
      int[] nums, int k
   )
   {
      var counter = new Dictionary<int, int>();

      foreach (int num in nums)
      {
         if (!counter.ContainsKey(num))
         {
            counter[num] = 0;
         }
         counter[num]++;
         if (counter[num] >= k)
         {
            return true;
         }
      }

      return false;
   }
}